### DISCUSSION TITLE
Guide to ai-deck development with toolbelt.

### DISCUSSION THREAD

**Discussion by Laatikk01234** [bitcraze/discussions#76] [Q&A]
Hi,

I am trying to add pthread library of freertos into the aideck example provided. 
Is there a way that this is intended to be done or is stabbing the docker container the right way to go about this?

->  A more comprehensive guide to toolbelt would be much appreciated.

**Comment by krichardsson** [comment]
If you want to modify the build environment you should probably add it to the docker image. You can start from https://github.com/bitcraze/docker-aideck and extend it as needed. 

The way the toolbelt is designed, it will unfortunately not be able to run your extended docker image as a builder, but you can run it directly from docker instead. 
You will get the full docker command that the toolbelt runs if you run a command it with the `-d` flag.

On my machine for instance, I get 
```
tb -d make
...
Running: docker run --rm -u 1000 -v /var/run/docker.sock:/var/run/docker.sock -it -v /home/kristoffer/code/bitcraze/aideck-gap8-examples:/module bitcraze/aideck tools/build/make
```

You can now simply copy/past the docker part and substitute `bitcraze/aideck` for your own image
